### ISSUE TITLE
Safari: weight tracker app has parse error after open, save, reopen

### ISSUE THREAD

**Issue by JonFerraiolo** [maqetta/maqetta#3927]
@peller @billreed63 @jhpedemonte 

If you download the "final version" of part 2 from this URL:

http://www.ibm.com/developerworks/apps/download/index.jsp?contentid=860909&filename=maqetta_part2.zip&method=http&locale=

and then:
- create a new project
- upload (and expand) the ZIP into your new project
- double-click on index.html to open it
- immediately Save, close, reopen

RESULT=> file fails to parse - error message appears at top of the window

**Comment by JonFerraiolo** [comment]
So far, I can only make this happen on Safari when running on maqetta-test. Likely a timing issue.

**Comment by JonFerraiolo** [comment]
Maybe related to #3584 and/or #3423 - the stack trace shows that an exception within _DatePickerMixin.js (which calls date.js). I wouldn't be surprised if the problem went away if we changed Maqetta to define mobile themes via a separate SCRIPT tag instead of extra properties on the SCRIPT tag for dojo.js.

Here is a picture of the stack trace in Safari Web Inspector:
![Screen Shot 2013-04-21 at 1 01 09 PM](https://f.cloud.github.com/assets/706721/406819/5a9ac048-aabe-11e2-9fb9-98d1729d8778.png)

@billreed63 @peller @jhpedemonte - Seems like we will have to defer this one.

**Comment by JonFerraiolo** [comment]
-> @peller

**Comment by peller** [comment]
FWIW, I only see this on Safari.  As shown on the console:

```
'null' is not an object (evaluating '_4.getMonth')

getDaysInMonth@http://maqetta-app-test.austin.ibm.com/maqetta/user/E/ws/workspace/proj5/lib/dojo/dojo/date.js:11
onDaySet@http://maqetta-app-test.austin.ibm.com/maqetta/user/E/ws/workspace/proj5/lib/dojo/dojox/mobile/_DatePickerMixin.js:73
reset@http://maqetta-app-test.austin.ibm.com/maqetta/user/E/ws/workspace/proj5/lib/dojo/dojox/mobile/_DatePickerMixin.js:56
startup@http://maqetta-app-test.austin.ibm.com/maqetta/user/E/ws/workspace/proj5/lib/dojo/dojox/mobile/_PickerBase.js:17
_2cf@http://maqetta-app-test.austin.ibm.com/maqetta/user/E/ws/workspace/proj5/lib/dojo/dojo/dojo.js:3023
startup@http://maqetta-app-test.austin.ibm.com/maqetta/user/E/ws/workspace/proj5/lib/dojo/dojox/mobile/SpinWheel.js:11
@http://maqetta-app-test.austin.ibm.com/maqetta/user/E/ws/workspace/proj5/lib/dojo/dijit/dijit.js:550
forEach@http://maqetta-app-test.austin.ibm.com/maqetta/user/E/ws/workspace/proj5/lib/dojo/dojo/dojo.js:6798
startup@http://maqetta-app-test.austin.ibm.com/maqetta/user/E/ws/workspace/proj5/lib/dojo/dijit/dijit.js:553
```

**Comment by peller** [comment]
This does seem to be a failure of the patch provided in #3584.  The patch is in place on the prototype -- I stepped into it -- but because the _started flag is true, it returns `[null, null, null]` instead of a valid date triple.  The original workaround, to provide a fixed date in all cases, even after startup, could avoid this bug but would produce the wrong date.

Given that this fails only in Safari and that this faulty widget is fixed in 1.9, I'm not sure it's worth messing around any further at this point.

**Comment by peller** [comment]
confirmed that it works in latest build
